store/store.go

```diff
@@ -899,7 +899,36 @@ func (s *Store) newRequest(reqOptions *requestOptions, user *auth.UserState) (*h
 	req.Header.Set("X-Ubuntu-Series", s.series)
 	req.Header.Set("X-Ubuntu-Classic", strconv.FormatBool(release.OnClassic))
 	req.Header.Set("X-Ubuntu-Wire-Protocol", UbuntuCoreWireProtocol)
+	// still send this for now
 	req.Header.Set("X-Ubuntu-No-CDN", strconv.FormatBool(s.noCDN))
+	if s.noCDN {
+		req.Header.Set("Snap-CDN", "none")
+	} else {
+		if s.authContext != nil {
+			// set Snap-CDN from cloud instance information
+			// if available
+			// TODO: do we want a more complex retry strategy
+			// where we first to send this header and if the
```

Comment:
where we first _attempt_ to send this header?

Reply:
a problem with this whole thing is that snapd doesn't talk to the CDN directly, it only gets bounced to it from the store when downloading, so tracking whether it works or not is rather tricky.

Reply:
yes, we would probably need to be control to follow 302 more explicitly at least for downloads, and distinguish errors trying to connect, starting a response, vs middle of response